sip/sip-001-burn-election.md

```diff
@@ -0,0 +1,1074 @@
+  to submit a cheap commitment and a lot of user burns.  This is
+important because it should not be possible for a leader to profit more from
+adaptively increasing their burns in response to other leader's burns.
+
+The first property is enforced by the reward distribution rules (see below),
+whereby a burn only receives a reward if its block successfully extended the
+"canonical" fork.  The second property is given "for free" because the underlying burn chain
+assesses each participant a transaction fee.  Users and leaders incur an ever-increasing
+cost of trying to adaptively out-burn other leaders by submitting more and more
+transactions.
+
+Peers who want to support a leader candidate must send their burn transactions _in the
+same burn chain block_ as the commitment transaction.  This limits the degree to
+which peers can adaptively out-bid each other to include their
+commitments.  However, this constraint creates an undesirable
+negative feedback loop for supporting
+leaders -- if there is _too much_ interest in a leader, then peers may accidentally 
+kick their preferred commitment out of the target burn block (or kick all
+burn commitments out).  Because a commitment transaction may only be accepted if
+it is mined in one block, accidentally bumping it out of the leader's targeted
+block will waste everyone's cryptocurrency and dissuade
+users from supporting a leader in the future.
+
+While there is no way to prevent this "burn collapse"
+scenario outright, the Stacks protocol helps reduce the likelihood of it
+happening through two mechanisms.  First, Stacks does not implement an all-or-nothing reward system --
+peers can hedge their bets by committing to 
+a distribution of chain tips, so they only need to send one burn transaction per chain tip candidate
+in order to receive _some_ reward instead of
+feeling compelled to one-up each other.  Second, the fork selection rules
+(described below) ensure that burns can only help leader so much -- they stop helping after a certain
+threshold, and too many burns over a short time hurts the leader's chances in future
+elections.
+
+# Reward distribution
+
+New Stacks tokens come into existence on a fork in an epoch where a leader is
+selected, and are granted to the leader if the leader produces a valid block.
+However, the Stacks blockchain pools all tokens created and all transaction fees received and
+does not distribute them until a large number of epochs (a _lockup period_) has
+passed.  The tokens cannot be spent until the period passes.
+
+The idealized payout distribution in a proof-of-burn blockchain is a
+distribution where each burner receives tokens proportional to the fraction of burns they
+contributed for the block.  In addition to implementing a lockup period, the
+Stacks protocol "smooths out" the token rewards over a _reward window_
+to help realize a close approximation of this payout distribution.
+
+## Sharing the rewards among winners
+
```

Comment:
Is this example a correct interpretation of this reward sharing scheme:
if there are three epochs, and a "reward window" of size 2, at time t = n + 1,
the first window rewards:

 the leader of Epoch 1 proportional to their burns over Epoch 1 and Epoch 2
 the leader of Epoch 2 proportional to their burns over Epoch 1 and Epoch 2

And the amount being rewarded here is the coinbase and the
transaction fee ?  I think for this to work with the way that the
40/60 split works with streamed transactions, we'd need a more specific scheme.
Because if a miner derives equalish benefit from
fees in the next block, they're incentivized to try and reorg a
microfork for heavy transactions, so that they get the 40% _and_
the 60%. I think we need it to be the case that a leader of epoch
n receives _no reward_ for the following epochs, so a window
could still be rewarded, but it'd be something like, the rewards
for Epoch N are distributed amongt the leaders of Epoch N, Epoch
N+1,... Epoch N+k.

Reply:
I like that -- the rewards for epoch _n_ are distributed amongst leaders of epochs _n_.._n+k_.  However, I think it might be safer if transaction fees for streamed transactions are _not_ distributed via a reward window, but are instead given directly to the leader in a 40/60 split.  What do you think?